### PR TITLE
Added queries to GET /api/reviews

### PR DESCRIPTION
**Pull request by doublN**
Added queries to GET /api/reviews including tests for sorting by a valid category, responding in ascending order, handling an invalid sort_by query, handling an invalid order query, and querying by an non-exsistant category.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -46,7 +48,11 @@ app.use("/*", (req, res, next) =>{
 app.use((err, req, res, next) =>{
     if(err.code === '22P02'){
         res.status(400).send({msg : "Bad request: invalid data type"})
-    } else{
+    } else if(err.code === '23503'){
+        res.status(404).send({msg : "Bad request: invalid data"})
+    }else if(err.code == '23502'){
+        res.status(400).send({msg : 'Bad request: missing data'})
+    }else{
         next(err);
```
Comment (doublN) [inline app.js:56]: Sorry, I think this was missed in the last pull request.

**Review by hevdevs — CHANGES_REQUESTED** [review]
Just need another 200 test here for when a category exists but has no reviews (i.e. children's games) in which case we'd want to get an empty array as a response

File: __tests__/reviews.test.js [hunk 2]
```diff
@@ -206,4 +206,106 @@ describe("GET /api/reviews", () =>{
             })
         })
     })
+
+    test("status 200: results sorted by title descending", () =>{
```
Comment (Rosem10) [inline __tests__/reviews.test.js]: I don't think you need to write a test for all of the possible sort_by options. Probably just one or two that are valid, and then one that isn't.

File: __tests__/reviews.test.js [hunk 3]
```diff
@@ -206,4 +206,106 @@ describe("GET /api/reviews", () =>{
...
+    test("status 200: results sorted by comment_count descending", () =>{
+        return request(app)
+        .get("/api/reviews?sort_by=comment_count")
+        .expect(200)
+        .then(({body}) =>{
+            expect(body.reviews).toBeSortedBy("comment_count", {descending : true})
+        })
+    })
+
+    test("status 200: results arrive in ascending order", () =>{
```
Comment (Rosem10) [inline __tests__/reviews.test.js]: Could do with a little clarification in the descriptions about the difference between this test and the one above/ make it explicit that desc is the default but asc would need to be passed in.

**Review by Rosem10 — APPROVED** [review]

**Pull request merged**